src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/GetAzureCognitiveServicesCommitmentPlanAssociation.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.CognitiveServices;
+using Microsoft.Azure.Management.CognitiveServices.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Management.CognitiveServices
+{
+    /// <summary>
+    /// Get Cognitive Services Account by name, all accounts under resource group or all accounts under the subscription
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesCommitmentPlanAssociation", DefaultParameterSetName = DefaultParameterSet), OutputType(typeof(PSCognitiveServicesAccount))]
+    public class GetAzureCognitiveServicesCommitmentPlanAssociationCommand : CognitiveServicesAccountBaseCmdlet
+    {
+        protected const string DefaultParameterSet = "DefaultParameterSet";
+        protected const string ResourceIdParameterSet = "ResourceIdParameterSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = false,
+            ParameterSetName = DefaultParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "Cognitive Services CommitmentPlan Name.")]
+        [ValidateNotNullOrEmpty]
+        public string CommitmentPlanName { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = false,
+            ParameterSetName = DefaultParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services CommitmentPlan AssociationName Name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            RunCmdLet(() =>
+            {
+                switch (ParameterSetName)
+                {
+                    case ResourceIdParameterSet:
+                        if (!CognitiveServices.ResourceId.TryParse(ResourceId, out CognitiveServices.ResourceId resourceId))
+                        {
+                            WriteError(new ErrorRecord(new Exception("Failed to parse ResourceId"), string.Empty, ErrorCategory.NotSpecified, null));
```

Comment:
```suggestion
                            throw new AzPsArgumentException ("Failed to parse ResourceId");
```
Recommend throwing AzPsExpection directly. See [here](https://microsoft.sharepoint.com/teams/IoTToolingTeam/_layouts/15/Doc.aspx?sourcedoc={8a3ef76b-61e5-4d4c-9ee5-69ab3a65816d}&action=edit&wd=target%28AZ%20PowerShell%2FKnowledge%20Transfer.one%7C74a747ad-7d88-4de3-8c49-d4416020404e%2FAzure%20PowerShell%20Exceptions%7Ce11b0dc9-d2ee-46a3-aba2-4d408c13aa68%2F%29&wdorigin=703) to know AzPsException more